docs/development/doc_guide.rst

```diff
@@ -1,76 +1,633 @@
+
+      warnings.warn("The subtract function encountered a nan value.", UserWarning)
+
+      return b - a if switch_order else a - b
+
+Template docstring
+~~~~~~~~~~~~~~~~~~
+
+This template docstring may be copied into new functions. Usually only
+some of the sections will be necessary for a particular function, but
+any sections that are included should be in the order provided.
+
+.. code-block:: python
+  :caption: Docstring template
+
+  def sample_function():
+      r"""
+      Return ...
+
+      Parameters
+      ----------
+
+      Returns
+      -------
+
+      Raises
+      ------
+
+      Warns
+      -----
+
+      See Also
+      --------
+
+      Notes
+      -----
+
+      References
+      ----------
+
+      Examples
+      --------
+
+      """
+
+
+Documentation guidelines
+========================
+
+This section contains guidelines and best practices for writing
+PlasmaPy documentation.
+
+* Write documentation to be understandable to students taking their
+  first course or beginning their first research project in plasma
+  science. Include highly technical information only when necessary.
+
+* Use the `active voice <https://en.wikipedia.org/wiki/Active_voice>`_
+  in the present tense.
+
+* Keep the documentation style consistent within a file or module, and
+  preferably across all of PlasmaPy's documentation.
+
+* Refer to the `numpydoc`_ standard for how to write docstrings for
+  classes, class attributes, and constants.
+
+* Update code and corresponding documentation at the same time.
+
+* Write sentences that are simple, concise, and direct rather than
+  complicated, vague, or ambiguous. Prefer sentences with ≲ 20
+  words.
+
+* Avoid idioms, metaphors, and references that are specific to a
+  particular culture.
+
+* Use technical jargon sparingly. Define technical jargon when
+  necessary.
+
+* Many words and software packages have more than one common spelling
+  or acronym. Use the spelling that is used in the file you are
+  modifying, which is preferably the spelling used throughout
+  `PlasmaPy's online documentation`_.
+
+  * More generally, it is preferable to use the spelling that is used
+    in `Python's documentation`_ or the spelling that is used most
+    commonly.
+
+  * Represent names and acronyms for a software package as they are
+    represented in the documentation for that package.
+
+* Write the full namespace when referring to code objects within
+  PlasmaPy. For example, write
+  ``~plasmapy.formulary.parameters.Alfven_speed`` rather than
+  ``~plasmapy.formulary.Alfven_speed``.  This does not need to be done
+  for intersphinx links to other packages; it is fine to do
+  ``~astropy.units.Quantity`` and not necessary to do
+  ``~astropy.units.quantity.Quantity``.
+
+* For readability, limit documentation line lengths to ≲ 72 characters.
+  Longer line lengths may be used when necessary (e.g., for hyperlinks).
```

Comment:
```suggestion
* For readability, limit documentation line lengths to ≲ 80 characters.
  Longer line lengths may be used when necessary (e.g., for hyperlinks).
```

This number comes from Python's PEP8 rule based on a max line length of 80.  Our max line length is 90 so I say we bump this up by 8-10 characters as well.

Reply:
I looked up recently what the [optimal line length](https://baymard.com/blog/line-length-readability) is for readability and I've seen ∼50–60 characters as well as ∼50–75 characters.  Perhaps we could say ≲ 72 characters excluding indentations.  That'd typically be a max line length of 76 characters for regular functions, so it ends up being an average of our two preferences.